File: write_ranking.js
const fs = require('fs')
const jsx = `import { useState, useEffect, useCallback, useRef } from 'react'
import { motion, AnimatePresence } from 'framer-motion'
import {
  Trophy, RefreshCw, Crown, Star,
  ChevronLeft, ChevronRight,
  CheckCircle, Clock, AlertTriangle, X,
  GraduationCap, Shield,
} from 'lucide-react'
import { siswaApi } from '../../services/siswaService'
import { BadgeOverlay } from '../../components/GachaHarian'
`
fs.writeFileSync('src/pages/siswa/Ranking.jsx', jsx, 'utf8')
console.log('written', jsx.length)

File: write_ranking.py
p = r"src/pages/siswa/Ranking.jsx"
lines = []
def w(*args): lines.extend(args)

w("import { useState, useEffect, useCallback, useRef } from " + chr(39) + "react" + chr(39))
w("import { motion, AnimatePresence } from " + chr(39) + "framer-motion" + chr(39))
w("import {")
w("  Trophy, RefreshCw, Crown, Star,")
w("  ChevronLeft, ChevronRight,")
w("  CheckCircle, Clock, AlertTriangle, X,")
w("  GraduationCap, Shield,")
w("} from " + chr(39) + "lucide-react" + chr(39))
w("import { siswaApi } from " + chr(39) + "../../services/siswaService" + chr(39))
w("import { BadgeOverlay } from " + chr(39) + "../../components/GachaHarian" + chr(39))
w("")
open(p, "w", encoding="utf-8").write("\n".join(lines))
print("ok lines=" + str(len(lines)))
